odoo/service/server.py

```diff
@@ -1276,10 +1276,10 @@ def load_test_file_py(registry, test_file):
     # pylint: disable=import-outside-toplevel
     from odoo.tests import loader  # noqa: PLC0415
     from odoo.tests.suite import OdooSuite  # noqa: PLC0415
-    threading.current_thread().testing = True
+    modules.module.current_test = True
     try:
         test_path, _ = os.path.splitext(os.path.abspath(test_file))
-        for mod in [m for m in get_modules() if '%s%s%s' % (os.path.sep, m, os.path.sep) in test_file]:
+        for mod in [m for m in modules.get_modules() if f'{os.path.sep}{m}{os.path.sep}' in test_file]:
```

Comment:
This change is not needed. We can still import get_modules

If the goal is clarity it is possible to have a better version IMHO 
```suggestion
        for mod in [m for m in modules.get_modules() m in test_file.split(os.path.sep)]:
```

But in any case I wouldn't change that, we will most likely remove it in the future (or at least I would like to). It is already possible to do something similar to test file using test tags.